serializer.go

```diff
@@ -173,13 +219,47 @@ func (s *Serializer) Write32BytesArraySlice(data SliceOfArraysOf32Bytes, lenType
 }
 
 // Write64BytesArraySlice writes a slice of arrays of 64 bytes to the Serializer.
-func (s *Serializer) Write64BytesArraySlice(data SliceOfArraysOf64Bytes, lenType SeriSliceLengthType, errProducer ErrProducer) *Serializer {
+func (s *Serializer) Write64BytesArraySlice(producer WriteSliceOfArraysOf64BytesProducerFunc, deSeriMode DeSerializationMode, lenType SeriSliceLengthType, arrayRules *ArrayRules, errProducer ErrProducer) *Serializer {
 	if s.err != nil {
 		return s
 	}
-	_ = s.writeSliceLength(len(data), lenType, errProducer)
+
+	data := producer()
+	sliceLength := len(data)
+
+	var arrayElementValidator ElementValidationFunc
+	if arrayRules != nil && deSeriMode.HasMode(DeSeriModePerformValidation) {
+		if err := arrayRules.CheckBounds(uint16(sliceLength)); err != nil {
+			s.err = errProducer(err)
+			return s
+		}
+
+		switch arrayRules.ValidationMode {
+		case ArrayValidModeNoValidation:
+		case ArrayValidModeDuplicates:
+			arrayElementValidator = arrayRules.ElementUniqueValidator()
+		case ArrayValidModeLexicalOrdering:
+			arrayElementValidator = arrayRules.LexicalOrderValidator()
+		case ArrayValidModeDuplicates | ArrayValidModeLexicalOrdering:
+			arrayElementValidator = arrayRules.LexicalOrderWithoutDupsValidator()
+		default:
+			panic(ErrUnknownArrayValidationMode)
+		}
+	}
+
+	_ = s.writeSliceLength(sliceLength, lenType, errProducer)
 	for i := range data {
-		if _, err := s.buf.Write(data[i][:]); err != nil {
+		element := data[i][:]
+
+		// check array element validation against previous element
```

Comment:
This comment is scattered in tests and here but it isn't always applicable. Maybe just remove it as it isn't correct.
For example, if only uniqueness is validated, the validator validates not just against the previous element. You know what I mean.